README.md

```diff
@@ -30,22 +28,50 @@ Use the `-port` option to specify the port number for the server to listen reque
 ./code-runner-server -port <port>
 ```
 
-## Send code files and arguments to server
-Use curl command to connect to the server and send request.
-Use multipart/form data option(-F) to provide the assignment submission file and the command-line arguments that are to be used when executing the code.
+## Code Runner as a docker image
+- A docker image of code runner is used as a base image for different language specific images.
+- Code Runner is not directly run, instead use the desired language specific image to run the assignments.
+- Prerequisite for this is that docker engine should be installed.
+- See [instructions](https://docs.docker.com/engine/installation/) for installing docker engine on different supported platforms.
+
+Use the following docker command to run the language specific image.
 ```commandline
-curl <server_ip_address:port_number> -F <compressed_filepath> -F <arg1> -F <arg2> ...
+docker run --publish <port_to_expose>:<port_to_run> <language_image_tag> -port <port_to_run>
 ```
 
+## Supported Languages
+A fixed set of languages and their versions are currently supported.
+- gcc 7
+- g++ 7
+- python 3.7
+- java 8 & 11
+
 ## Web page to Submit Assignment
-![](docs/codeRunnerPage.png)
+Use the exposed port number to open the web page. Example - for exposed port number as 52453, hit `localhost:52453` on browser.
+
+![](docs/webPage.png)
 
 Submit your assignment through the web page as shown in the figure above.
+- Initially `Build` and `Run` buttons are disabled. Submit the assignment details to enable them.
```

Comment:
what details?

Reply:
Details as in the tarball, commands, working directory etc. But I guess I should simply say 'Submit the assignment to enable them.'